### PR TITLE
Fix a bug in List base type when used in verdi shell

### LINKED ISSUE #1171: Bug in List  base data type
It is not possible to store an empty list. To reproduce the error one needs to do the following: 

```
In [1]: from aiida.orm.data.base import List
In [2]: l = List()
In [3]: l.store()
```

The error message is the following:

> 
> /home/epfl/work/aiida_core/aiida/orm/data/__init__.pyc in _set_attr(self, key, value, **kwargs)
>     115         if self.is_stored:
>     116             raise ModificationNotAllowed(
> --> 117                 "Cannot change the attributes of a stored data node.")
>     118         super(Data, self)._set_attr(key, value, **kwargs)
>     119 
> 
> ModificationNotAllowed: Cannot change the attributes of a stored data node.

After that any attempt to access the stored empty list is followed by exactly the same error:
```
In [5]: print l
```

However, if an empty List is accessed at least one time before the storage  - everything works correctly

```
In [1]: from aiida.orm.data.base import List
In [2]: l = List()
In [3]: print l
[]
In [4]: l.store()
Out[4]: <List: []>
In [5]: print l
[]
```

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #1171 

Instantiating a new List instance and storing it before calling any method
that would set the internal list would trigger a ModificationNotAllowed
exception. That is to say the following in verdi shell would except:

	from aiida.orm.data import List
	l = List()
	l.store()

This has to do with the fact that the internal list, stored in an attribute,
would not be created until it was accessed. This was done in the _get_list()
However, in this example when calling store, the internal set was not yet
defined as it was stored and afterwards ipython tried to pretty print it
invoking repr() which tried to get the internal list, which didn't exist.

Long story short, this was a bug that only appeared in this way in verdi
shell. For that reason I could not write a test that reproduced the problem.

### CODE REVIEW

File: aiida/orm/data/__init__.py [hunk 1]
```diff
@@ -471,6 +471,10 @@ class List(Data, MutableSequence):
     """
     _LIST_KEY = 'list'
 
+    def __init__(self):
+        super(List, self).__init__()
+        self._set_list(list())
```
Comment (ltalirz) [inline aiida/orm/data/__init__.py]: There is no way to initialize the list via constructor parameters, right?
If there was, this would be a problem, since this code forces the list to always be empty at construction.
An alternative might be to `_set_list` before calling the superclass constructor

Comment (giovannipizzi) [reply aiida/orm/data/__init__.py]: Still I think that one should accept kwargs params and pass them to the super.__init__ (e.g. if reloading from a UUID). Actually, with this change, would load_node(some_uuid) work, where some_uuid is the uuid of a previously stored List?

Comment (sphuber) [reply aiida/orm/data/__init__.py]: No that does not work. We can pass args and kwargs to the `Data` constructor like you suggest, but do you also want a way to do something like `l = L([1,2,3])`?

Comment (giovannipizzi) [reply aiida/orm/data/__init__.py]: No, as you say this is not supported, but as there is a _set_list, this should work (after passing the *args and **kwargs, and putting the logic below in some kind of `if` that avoids setting it to an empty list if it was already set): 
`L(_list=[1,2,3])`, similarly to `ParameterData(dict={})`.
Even better, if there is no real reason to keep _set_list private, one can rename it to `set_list` and then it would be `L(list=[1,2,3])`.
